src/k8s-engine/lima.ts

```diff
@@ -926,6 +984,11 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
           100,
           this.client?.waitForReadyNodes() ?? Promise.reject(new Error('No client')));
 
+        // The socket isn't present immediately after `docker start` finishes, but it should be by this point
```

Comment:
This comment sets off alarm bells for me; it basically says: "we have a race condition here, but I think we will be fine".

I think the proper way should be to wait until the socket is available, and *then* change permissions.

I wonder if this could be done inside the provisioning script:

```shell
rc-service docker start
while ! [ -S /var/run/docker.sock ]; do sleep 1; done
chmod 666 /var/run/docker.sock
```

Reply:
Yes, that sounds better.  I ssh'ed in and ran a `watch ls /run` command and it was created shortly after the server was started, well before the K8s API was ready, but scripting it would be better.